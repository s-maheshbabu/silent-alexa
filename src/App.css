/* TODO The App related CSS should be deleted */
.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/** Scaffolding **/
#page {
  display: grid;
  grid-template-columns: 1fr 300px;
  grid-template-rows: 30px auto 30px;
  height: 100vh;
}

/** Panel Module **/

/** TODO: Right panel should also be moved into its own CSS file once the right panel component gets created. **/
#rightpanel {
  display: grid;
  grid-template-rows: auto;
  height: calc(100vh - 60px);
}

.panel-body {
  overflow: auto;
}

/** Header **/
#header {
  grid-row: 1;
  grid-column: 1 / 3;

  display: grid;
  grid-template-columns: 1fr auto;
}

#header-info {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

  display: flex;
  justify-content: center;
}

#header-controls {
  display: flex;
  height: 100%;
  justify-content: flex-end;
}

/** Footer **/

#footer {
  grid-column: 1 / 3;
  grid-row: 3;

  display: grid;
  grid-template-columns: 1fr auto;
}

#footer-info {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

#footer-controls {
  display: flex;
  height: 100%;
  justify-content: flex-end;
}

#footer-controls a {
  height: 100%;
  display: flex;
  align-items: center;
}

/** Styles **/
body {
  margin: 0;
  padding: 0;

  background-color: #d7dae0;
  font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto,
    "Helvetica Neue", Arial, sans-serif;
  font-size: 0.8rem;
  color: #2c313a;
}

#header {
  align-items: center;
  border-top: 1px solid #c5cad3;
}

#footer {
  align-items: center;
  border-top: 1px solid #c5cad3;
}

#footer-controls a {
  padding: 0 10px;
  color: initial;
  text-decoration: none;
}

#footer-controls a:hover {
  background-color: #c5cad3;
}
