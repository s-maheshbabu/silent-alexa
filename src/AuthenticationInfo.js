import { hasIn } from "immutable";
import util from "util";

export default class AuthenticationInfo {
  constructor(authResponse) {
    if (arguments.length == 1 && this.validateAuthResponse(authResponse)) {
      var _access_token = authResponse.access_token;
      var _expires_in = authResponse.expires_in;
    }
    this.getAccessToken = function() {
      return _access_token;
    };
  }

  isValid() {
    return this.getAccessToken() ? true : false;
  }

  validateAuthResponse(authResponse) {
    if (
      hasIn(authResponse, ["access_token"]) &&
      hasIn(authResponse, ["expires_in"]) &&
      authResponse.access_token &&
      authResponse.expires_in
    ) {
      return true;
    } else {
      console.log(
        "Encountered an error on login: " +
          util.inspect(authResponse, { showHidden: true, depth: null })
      );
      return false;
    }
  }
}
