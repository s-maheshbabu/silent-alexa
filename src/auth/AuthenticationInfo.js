import IllegalArgumentError from "errors/IllegalArgumentError";
import Cookies from "js-cookie";
import util from "util";

/**
 * Wrapper class to hold access_token and expires_in returned in LoginWithAmazon
 * authorization response.
 * Instances are created only if access_token and expires_in are defined. The token
 * is stored in a cookie called 'amazon_Login_accessToken' because that is the
 * cookie that LoginWithAmazon will look for.
 * https://developer.amazon.com/docs/login-with-amazon/javascript-sdk-reference.html#setUseCookie
 */
// TODO: Rename this to AuthenticationManager now that the class has been refactored to not
// actually hold the authentication information.
class AuthenticationInfo {
  persist(lwaResponse) {
    const numberOfSecondsInADay = 86400;

    if (lwaResponse && lwaResponse.access_token && lwaResponse.expires_in) {
      Cookies.set(AMAZON_LOGIN_COOKIE, lwaResponse.access_token, {
        expires: lwaResponse.expires_in / numberOfSecondsInADay,
        secure: false // TODO: Change localhost to also use https and then change this to true.
      });
    } else {
      const serializedLWAResponse = util.inspect(lwaResponse, {
        showHidden: true,
        depth: null
      });
      throw new IllegalArgumentError(
        `LoginWithAmazon Authorization response is undefined or 
          doesnt have access_token/expires_in.
          lwaResponse: ${serializedLWAResponse}`
      );
    }
  }

  /**
   * @returns true is authentication info is present and false otherwise.
   * This does not guarantee that the tokens haven't already expired.
   * Callers should be able to handle the case of tokens being
   * invalid when they try to use them.
   */
  isPresent = () => {
    return this.getAccessToken() !== undefined;
  };

  /**
   * @returns access_token to access Alexa API
   */
  getAccessToken = () => {
    return undefined;
  };

  /**
   * @returns deleted the cookie that stores the authentication info
   */
  clear = () => {
    Cookies.remove(AMAZON_LOGIN_COOKIE);
  };
}

// Declare a singleton
const instance = new AuthenticationInfo();
Object.freeze(instance);

export default instance;
export const AMAZON_LOGIN_COOKIE = "amazon_Login_accessToken";
