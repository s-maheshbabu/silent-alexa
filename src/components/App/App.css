/** Scaffolding **/
#page {
  display: grid;
  grid-template-columns: 1fr 300px;
  grid-template-rows: 60px auto 30px;
  height: 100vh;
}

/** Styles **/
body {
  margin: 0;
  padding: 0;

  background-color: #ffffff;
  font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto,
    "Helvetica Neue", Arial, sans-serif;
  font-size: 0.8rem;
  color: #2c313a;
}
