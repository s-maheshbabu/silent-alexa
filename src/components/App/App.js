import React, { Component } from "react";
import "./App.css";

import Header from "Header/Header";
import Body from "Body/Body";
import Footer from "Footer/Footer";

class App extends Component {
  render() {
    return (
      <div id="page">
        <Header />
        <Body />
        <Footer />
      </div>
    );
  }
}

export default App;
