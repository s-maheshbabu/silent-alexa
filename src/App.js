import React, { Component } from "react";
import "./App.css";
import util from "util";

import Header from "./Header";
import Body from "./Body";
import Footer from "./Footer";
import AuthenticationInfo from "./AuthenticationInfo";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      authenticationInfo: undefined
    };
  }
  render() {
    return (
      <div id="page">
        <Header
          updateAuthenticationInfo={authenticationInfo =>
            this.updateAuthenticationInfo(authenticationInfo)
          }
        />
        <Body authenticationInfo={this.state.authenticationInfo} />
        <Footer />
      </div>
    );
  }

  /**
   * Updates authenticationInfo prop in the component's state
   * @param authenticationInfo Instance of AuthenticationInfo
   */
  updateAuthenticationInfo = function(authenticationInfo) {
    if (
      authenticationInfo &&
      authenticationInfo instanceof AuthenticationInfo
    ) {
      this.setState({ authenticationInfo: authenticationInfo });
    } else {
      console.log(
        `Ignoring update. Received invalid authenticationInfo object: ${util.inspect(
          authenticationInfo,
          {
            showHidden: true,
            depth: null
          }
        )}.`
      );
    }
  };
}

export default App;
