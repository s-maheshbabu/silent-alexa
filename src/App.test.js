import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import AuthenticationInfo from "./AuthenticationInfo";

let app;
let appInstance;
let originalState;

beforeEach(() => {
  app = shallow(<App />);
  appInstance = app.instance();
  originalState = appInstance.state;
});

it("renders correctly without crashing", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
  wrapper.unmount();
});

it("verifies that authenticationInfo is passed to Body component", () => {
  const wrapper = shallow(<App />);
  const originalAuthenticationInfo = wrapper.instance().state
    .authenticationInfo;

  // Verify that Body recieves authenticationInfo property
  const authenticationInfoProp = wrapper
    .find("Body")
    .prop("authenticationInfo");
  expect(authenticationInfoProp).toBe(originalAuthenticationInfo);
});

it("verifies that updateAuthenticationInfo function is passed to Header component", () => {
  const updateAuthenticationInfoSpy = jest.spyOn(
    appInstance,
    "updateAuthenticationInfo"
  );

  // Verify that Header is passed on updateAuthenticationInfo property
  const headerUpdateAuthenticationInfoProp = app
    .find("Header")
    .prop("updateAuthenticationInfo");
  expect(headerUpdateAuthenticationInfoProp.length).toBe(1);

  // Verify that calling prop function passed to header calls updateAuthenticationInfo
  const someArgument = "dummyArgument";
  headerUpdateAuthenticationInfoProp(someArgument);
  expect(updateAuthenticationInfoSpy).toHaveBeenCalledWith(someArgument);
});

it("should not change state's authenticationInfo prop when authenticationInfo instance (implicit grant) is not valid", () => {
  const notAuthenticationInfoObject = { SomeObject: "not_authentication_info" };
  let undefinedAuthenticationInfo;

  const invalidAuthenticationInfoObjects = [
    notAuthenticationInfoObject,
    undefinedAuthenticationInfo
  ];

  const expectedState = appInstance.state;

  for (let i = 0; i < invalidAuthenticationInfoObjects.length; i++) {
    appInstance.updateAuthenticationInfo(invalidAuthenticationInfoObjects[i]);

    expect(appInstance.state).toEqual(expectedState);
  }
});

it("should change state when authenticationInfo instance (implicit grant) is defined and valid", () => {
  const lwaResponse = {
    access_token: "some_access_token",
    expires_in: 30
  };
  const expectedAuthenticationInfo = new AuthenticationInfo(lwaResponse);
  appInstance.updateAuthenticationInfo(expectedAuthenticationInfo);
  expect(appInstance.state.authenticationInfo).toBe(expectedAuthenticationInfo);
});
