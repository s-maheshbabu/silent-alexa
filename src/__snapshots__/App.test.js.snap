// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without crashing 1`] = `
<div
  id="page"
>
  <Header
    updateAuthenticationInfo={[Function]}
  />
  <Body />
  <Footer />
</div>
`;
